### DISCUSSION TITLE
[小調查] 關於威注音輸入法選字窗的取捨習慣。

### DISCUSSION THREAD

**Discussion by ShikiSuen** [vChewing/vChewing-macOS#462] [Polls]
向各位威注音輸入法的使用者問好。

截至發文時，威注音輸入法 for macOS 已經到了 v3.5.2 版。
（下文簡稱 MACV）

有個關乎到 MACV 今後發展前途的問題，
　　筆者想在這裡聽聽各位的意見。

先講一下問題背景。

筆者實在是有點想拋棄 IMK 選字窗了。
這玩意妨礙了威注音輸入法的跨平台模組化。
（模組化之後，可以方便移植到 iOS 和 Linux 等其他平台。）
目前威注音輸入法這樣子下去，總讓筆者覺得沒前途。
用「牛想往前繼續耕地，但被拽住了睪丸、
　　稍微動一下就很痛」形容的話比較恰當。

IMK 選字窗有很多缺陷，包括「Annotation API 空實作」在內。
筆者趁今年 WWDC 專門跟 Apple 約了專員
　　討論這事情，討論該怎樣應對這些問題。
然而，被安排的專員不是輸入法團隊的人，只是來捎口信安慰筆者的，
而且輸入法團隊現在被禁止回應任何與 InputMethodKit API 有關的提問。
（至少一個事實：Apple 員工被禁止私自對外承諾「某個故障是否會被解決」。）
至於 macOS 14 游標上下文輸入法選擇指示器，
Apple 到現在也沒給任何官方指示、來指導第三方開發者們該怎麼做。
筆者只能乾瞪眼，不得不承認 IMK 選字窗毫無功能拓展之可能性、可控性。

然而，留著 IMK 選字窗的原因是：

第一點：
IMK 選字窗的多行陣列陳列模式實在太經典，體驗獨一無二。
迄今為止，只有微信輸入法 for macOS 模仿得最像，
但微信輸入法在候選字詞特別多的時候，操作響應就會變得「遲鈍」。
（威注音自家的田所選字窗暫時不知該如何實現完美的捲動特性，很尬。）

第二點：
IMK 選字窗在 Intel Mac 電腦內的反應速度是最快的。
田所選字窗在 Intel Mac 電腦內使用時，
無論多行/多列顯示，無論是 Cocoa 版還是 SwiftUI 版，都
還是會有嚴重的效能缺陷。

筆者另有實作過一版「整個介面都只用一個
　　NSAttributedString 搞定」的前端形態（尚未公開），
卻又能做到比 IMK 選字窗還要快的響應速度了，
　　但這個模式卻無法實作滑鼠點擊。
（Cocoa 版與 SwiftUI 版可以做到
　「滑鼠左鍵點擊確認候選字」「滑鼠右鍵選單控頻或刪詞」）。

出於經濟等原因，筆者一直沒有任何條件親自
　　測試這個選字窗在 Apple Silicon 電腦上的效能。
（Apple Store 直營店不允許這麼做。）

於是這種事情越想越恐慌、越想越焦慮。
與其空想，不如聽聽各位使用者的意見。
就煩請各位在輸入法偏好設定內啟用/停用 IMK 選字窗，
然後分別用一段時間，再講一講自己的偏好。
這有助於方便筆者決定是否真的要割除 IMK 選字窗的支援。

先謝過各位了。

P.S.: 威注音最早期版本所使用的 Voltaire 選字窗不會再重新拿來用。
Voltaire 選字窗雖然由 Zonble 重寫過、被筆者改過美工，但骨子裡
　　仍是原作者 Lukhnos 在十二年前的 MVC 架構 + Cocoa 內容繪製邏輯。
一言以蔽之：筆者不知道 Voltaire 算不算 MVC 架構，
但目前威注音自家的田所選字窗用的 MVVM 架構與 MVC 完全是兩回事。
吃過 MVVM 甜頭的人，不會再輕易地想重新啟用難以控制改良的 MVC。

順頌時祺
Shiki Suen
2023-07-01

**Comment by larryjclai** [comment]
有機會模組化跨平台的話我覺得是好事，可以累積更多使用者

**Comment by ShikiSuen** [comment]
該調查已過期。
